File: python/jump_game.py
# no. 55
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        appoarchable = [0 for _ in range(len(nums))]
        appoarchable[0] = 1
        for i in range(len(nums)):
            if appoarchable[-1]:
                return 1
            if appoarchable[i]:
                for k in range(i + 1, i + nums[i] + 1):
                    if k >= len(nums):
                        break
                    if not appoarchable[k]:
                        appoarchable[k] = 1
        return appoarchable[-1]

File: C++/jump_game.cpp
// no. 55
#include <vector>
using namespace std;

class Solution {
public:
  bool canJump(vector<int> &nums) {
    vector<bool> appoarchable;
    for (int i = 0; i < nums.size(); i++) {
      appoarchable.push_back(false);
    }
    appoarchable[0] = true;
    for (int i = 0; i < nums.size(); i++) {
      if (appoarchable.back()) {
        return true;
      }
      if (appoarchable[i]) {
        for (int k = i + 1; k < nums[i] + i + 1; k++) {
          if (k >= nums.size()) {
            break;
          }
          if (!appoarchable.back()) {
            appoarchable[k] = true;
          }
        }
      }
    }
    return appoarchable.back();
  }
};
